rbi/stdlib/csv.rbi

```diff
@@ -63,34 +288,72 @@ class CSV < Object
   def self.read(path, options=T.unsafe(nil)); end
 end
 
-sig do
-  params(
-      io: T.any(::Sorbet::Private::Static::IOLike, String),
-      options: T::Hash[Symbol, T.untyped],
-  )
-  .returns(CSV)
-end
-
+# A CSV::Row is part Array and part Hash.  It retains an order for the fields
+# and allows duplicates just as an Array would, but also allows you to access
+# fields by name just as you could if they were in a Hash.
+#
+# All rows returned by CSV will be constructed from this class, if header row
+# processing is activated.
 class CSV::Row < Object
   include Enumerable
 
   extend T::Generic
   Elem = type_member(:out, fixed: T.nilable(String))
 end
 
+# A FieldInfo Struct contains details about a field's position in the data
+# source it was read from.  CSV will pass this Struct to some blocks that make
+# decisions based on field structure.  See CSV.convert_fields() for an example.
+#
+# **`index`**
+# :   The zero-based index of the field in its row.
+# **`line`**
+# :   The line of the data source this row is from.
+# **`header`**
+# :   The header for the column, when available.
+#
 class CSV::FieldInfo < Struct
   extend T::Generic
   Elem = type_member(:out, fixed: T.untyped)
 end
 
+# The error thrown when the parser encounters illegal CSV formatting.
 class CSV::MalformedCSVError < RuntimeError
 end
 
+# A CSV::Table is a two-dimensional data structure for representing CSV
+# documents.  Tables allow you to work with the data by row or column,
+# manipulate the data, and even convert the results back to CSV, if needed.
+#
+# All tables returned by CSV will be constructed from this class, if header row
+# processing is activated.
 class CSV::Table < Object
   include Enumerable
 
   extend T::Generic
   Elem = type_member(:out)
 end
 
+# Passes `args` to CSV::instance.
+#
+#     CSV("CSV,data").read
+#       #=> [["CSV", "data"]]
+#
+# If a block is given, the instance is passed the block and the return value
+# becomes the return value of the block.
+#
+#     CSV("CSV,data") { |c|
+#       c.read.any? { |a| a.include?("data") }
+#     } #=> true
+#
+#     CSV("CSV,data") { |c|
+#       c.read.any? { |a| a.include?("zombies") }
+#     } #=> false
+sig do
```

Comment:
I moved the `sig` here since it seemed to have been in the wrong place, plus I didn't know where the doc would go if the `sig` and `def` are separated like that.